druntime/src/core/sys/windows/winsock2.d

```diff
@@ -702,12 +702,12 @@ struct hostent
 // Note: These are Winsock2!!
 struct WSAOVERLAPPED;
 alias LPWSAOVERLAPPED = WSAOVERLAPPED*;
-alias LPWSAOVERLAPPED_COMPLETION_ROUTINE = void function(uint, uint, LPWSAOVERLAPPED, uint);
+alias LPWSAOVERLAPPED_COMPLETION_ROUTINE = void function(uint, uint, LPWSAOVERLAPPED, uint) nothrow @nogc;
```

Comment:
oh I forgot t finish the rest of my review. Could you please explain why this is OK, i.e. why you should never throw or GC allocate from such a callback?

Reply:
Re `nothrow`; since it's called from some internal OS context, there's no knowing what an exception escaping this callback into the OS might do, or where it may or may not be caught. I think it's absolutely appropriate to require that the user catches a D exception before returning to the operating system.

`@nogc` might be more questionable, but as I see it; if the callback may GC alloc, then essentially, the windows runtime itself is not `@nogc` since it's not really know-able from where or when it may call this callback...
I reckon it's safer to not do anything windows wouldn't normally do, but I guess `@nogc` could theoretically be removed...
What I would do, is add this attribute and then relax it if/when someone can show a compelling use case.

Reply:
fair